### PR TITLE
Add flows dynamically

### PR DESCRIPTION
**Pull request by dan2li**
[Feature] Support dynamically appending flows after the flow service started.

**Code Change:**
* Add interface `AppendFlows()`  to `FlowService` and `FlowRuntime`.
* Initialize queues for appended flows.

### CODE REVIEW

**Comment by s8sg** [comment]
Hey Dan, Lets create an issue first. We can discuss first and decide on the
approach and necessity.

**Comment by dan2li** [comment]
@s8sg  Here, https://github.com/s8sg/goflow/issues/75

**Review by s8sg — CHANGES_REQUESTED** [review]
I’ll suggest to update the register to do both. It will handle both when queue worker has started and we want to append

**Comment by dan2li** [comment]
@s8sg 
New commit to enable `Register()` to support dynamically registering flow.
Changes:
1.  Add `Register` method to the `FlowRuntime` to support dynamically registering flow.
2. Add `EnterWorkerMode` and `ExitWorkerMode` methods to the `FlowRuntime` to enable entering and exiting worker mode.
3. Remove `runtime.StartQueueWorker`, as it is no longer needed to create task queues for each flow at startup.

**Review by s8sg — APPROVED** [review]

File: v1/goflow.go [hunk 1]
```diff
@@ -178,95 +178,74 @@ func (fs *FlowService) Register(flowName string, handler runtime.FlowDefinitionH
...
+
 	errorChan := make(chan error)
 	defer close(errorChan)
-	if err := fs.initRuntime(); err != nil {
+	if err := fs.initRuntime(errorChan); err != nil {
 		return err
 	}
-	go fs.runtimeWorker(errorChan)
+
+	if err := fs.setWorkerMode(false); err != nil {
```
Comment (s8sg) [inline v1/goflow.go:226]: This will block ?

**Comment by s8sg** [comment]
@dan2li I faced some issue with the MR 
Created a new MR: https://github.com/s8sg/goflow/pull/79
We can discuss and work on the changes there 
Once everything looks good you can create a new MR. 
WDYT ?

**Comment by s8sg** [comment]
Also lets add some overview of the changes in terms on 
* How its impact the responsibility of functions 
* If possible add  a flowchart of the code

**Pull request merged**